### DISCUSSION TITLE
Eslint cant find the config file in the root folder of a project when it is a multi-folder workspace in VS Code... Any suggestions?

### DISCUSSION THREAD

**Discussion by PaulHaze** [eslint/eslint#18452] [Q&A] [answered]
So I have recently switched over to Eslint 9 and one thing I have noticed that I can NOT seem to fix is that it can only find the config file if I have ONLY that project open in VS Code. When I have a root directory with multiple projects in, then it only looks for the config in the top level directory that is open, not in the root of the current project I am working on.

For example, I have an 'Astro' workspace that I have all my astro projects in. When opened the the directory structure looks something  like this in VS Code:

<img width="306" alt="Screen Shot 2024-05-15 at 9 41 29 am" src="https://github.com/eslint/eslint/assets/20569604/1ad35e50-d6dc-4f1f-97c1-e1e69836d534">

When I open a  file from a folder like astro-tailwind-starter which has eslint 9 installed I get this error:

```
[Error - 09:43:03] An unexpected error occurred:
[Error - 09:43:03] Error: Could not find config file.
at locateConfigFileToUse (/Users/paulhayes/code/astro/starters-mine/astro-tailwind-starter/node_modules/.pnpm/eslint@9.2.0/node_modules/eslint/lib/eslint/eslint.js:350:21)
    at async calculateConfigArray (/Users/paulhayes/code/astro/starters-mine/astro-tailwind-starter/node_modules/.pnpm/eslint@9.2.0/node_modules/eslint/lib/eslint/eslint.js:385:49)
    at async ESLint.lintText (/Users/paulhayes/code/astro/starters-mine/astro-tailwind-starter/node_modules/.pnpm/eslint@9.2.0/node_modules/eslint/lib/eslint/eslint.js:1023:25)
    at async /Users/paulhayes/.vscode/extensions/dbaeumer.vscode-eslint-2.4.4/server/out/eslintServer.js:1:24954
    at async E (/Users/paulhayes/.vscode/extensions/dbaeumer.vscode-eslint-2.4.4/server/out/eslintServer.js:1:19114)
    at async /Users/paulhayes/.vscode/extensions/dbaeumer.vscode-eslint-2.4.4/server/out/eslintServer.js:1:220290
```

When I open the 'astro-tailwind-starter' on its own in VS Code it works fine. If I install eslint into the top level astro directory the error goes away, but I don't need anything installed in the top level folder!

Does anyone know how to get Eslint 9 to find my local config? I had this issue sometime in the past with Eslint + Typescript not finding their relative configs and I used to have put something like this into the eslintrc file:

```
	parserOptions: {
		tsconfigRootDir: __dirname,
		sourceType: 'module',
		ecmaVersion: 'latest',
	},
```

Do I need add something similar? I'm just confused why it works when its open on its own, but not when I have multiple projects open.

FWIW Eslint 8.57 which is being used in astro-tailwind-starter-old is working no problem in this set up...

**Comment by Rec0iL99** [comment, marked as answer]
Hi @PaulHaze, the ESLint core team does not maintain the VSCode extension for ESLint. Please contact the maintainers there for more help.

> When I have a root directory with multiple projects in, then it only looks for the config in the top level directory that is open, not in the root of the current project I am working on.

That's how the flat config system is designed. ESLint expects your config file to be in the root of the project and once it finds this file it stops searching for `eslint.config.js` files, unlike how it worked in the `eslintrc` config system where config files would be merged together. I believe the best way to solve this is to move all your individual ESLint configurations into a single root `eslint.config.js` and target specific project folders using the [`files`](https://eslint.org/docs/latest/use/configure/configuration-files#specifying-files-and-ignores) key.

[config file resolution doc](https://eslint.org/docs/latest/use/configure/configuration-files)
[vscode-eslint](https://github.com/microsoft/vscode-eslint)

**Reply by PaulHaze** [reply]
Is there any way to override this default behaviour and have Eslint look for a config in the project folder rather than the root? I often work with a root folder/workspace that has LOTS of projects and to install Eslint into the root of each workspace seems a little counter-intuitive, especially as there are linting rules within each project that are relevant to that project....

EDIT: Just looking a bit further into it, this seems to be more a VS Code problem than an Eslint problem (as they have a number of open issues describing my problem over at https://github.com/microsoft/vscode-eslint/issues so I will keep my eye open over there for a solution as well...

FURTHER EDIT: It seems that adding
```
{
  "eslint.workingDirectories": [
    {
      "mode": "auto"
    }
  ]
}
```
into a .vscode/settings.json file in the root of your project directory will cause the project-specific version of eslint.config.mjs to be loaded. This solution worked for me anyway so I will leave this here in case it helps someone else. I found the solution in this thread:
https://github.com/microsoft/vscode-eslint/issues/1826

**Reply by DamianoPantaniHT** [reply]
`"eslint.useFlatConfig": true` -> **`"eslint.useFlatConfig": false`** made it for me
